File: DAL/Models/Villes.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DAL.Models
{
    public class Villes:AuditableEntity
    {
        public Villes()
        {
            this.BankInfos = new HashSet<BankInfos>();
            this.Fournisseurs = new HashSet<Fournisseurs>();
        }

        public int Id { get; set; }
        public string Name { get; set; }

        //navigation
        public int PaysId { get; set; }
        public virtual Pays Pays { get; set; }

        
        public virtual ICollection<BankInfos> BankInfos { get; set; }
        public virtual ICollection<Fournisseurs> Fournisseurs { get; set; }
    }
}

File: Atomics Manager/ClientApp/src/app/admin/models/villes.ts
import { Pays } from './pays.model';

export class Villes {


    constructor(name?: string, paysName?: string, paysId?: string, pays?: Pays) {
        this.name = name;
        this.paysName = paysName;
        this.pays = pays;
        this.paysId = paysId;
    }

    public id: number;
    public name: string;
    public paysName: string;
    public paysId: string;
    public pays: Pays;

}
